src/log.c

```diff
@@ -55,54 +68,49 @@ void log_message_internal(const char* tag, log_level_t level, const char* messag
     fprintf(out, "\n");
     fflush(out);
 }
-void log_message_vargs(log_producer_t *tag, log_level_t level, const char *message, va_list args) {
-    for(int i =0; i<= log_service->size-1; ++i){
-        log_sink_t* sink = log_service->sinks[i];
-        if (level > sink->min_level) {
+
+void log_message_internal(log_producer_t *producer, log_level_t level, const char *message, va_list args) {
+    for(uint8_t i = 0; i < log_sinks_registered_count; ++i){
+        log_sink_t* sink = &log_sinks_registered_list[i];
```

Comment:
I think that you can remove that pointer.

Reply:
Fixed!